apps/webapp/src/components/staking/use-validator-infos.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  getValidatorInfo,
+  getVotingPowerFromValidatorInfo,
+  joinLoHiAmount,
+} from '@penumbra-zone/types';
+import { ValidatorInfo } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/component/stake/v1/stake_pb';
+import { useCollectedStream } from '../../fetchers/stream';
+import { getValidatorInfos } from '../../fetchers/staking';
+import { useMemo, useRef } from 'react';
+
+const byVotingPower = (validatorInfoA: ValidatorInfo, validatorInfoB: ValidatorInfo) =>
+  Number(
+    joinLoHiAmount(getVotingPowerFromValidatorInfo(validatorInfoB)) -
+      joinLoHiAmount(getVotingPowerFromValidatorInfo(validatorInfoA)),
+  );
+
+const toTotalVotingPower = (prev: number, curr: ValidatorInfo) =>
+  prev + Number(joinLoHiAmount(getVotingPowerFromValidatorInfo(curr)));
+
+const getFormattedVotingPower = (validatorInfo: ValidatorInfo, totalVotingPower: number) =>
+  Math.round(
+    (Number(joinLoHiAmount(getVotingPowerFromValidatorInfo(validatorInfo))) / totalVotingPower) *
+      100,
+  );
```

Comment:
looks like some more staking stuff that could move to a staking utils file in packages somewhere

Reply:
Done